users/dierkes/preprocessing/wav2vec.py

```diff
@@ -10,58 +11,206 @@ class FairseqAudioManifestCreationJob(Job):
     Creates required manifest files for wav2vec pretraining with fairseq. For the
     script see https://github.com/pytorch/fairseq/blob/main/examples/wav2vec/wav2vec_manifest.py
     """
+
+    __sis_hash_exclude__ = {
+        "upsampling_alpha": None,
+        "manifest_audio_paths": None,
+    }
+
     def __init__(
-        self, audio_dir_path, file_extension="wav", valid_percent=0.01, seed=42, path_must_contain=None
+        self,
+        audio_dir_path,
+        file_extension="wav",
+        valid_percent=0.01,
+        seed=42,
+        path_must_contain=None,
+        upsampling_alpha=None,
+        manifest_audio_paths=None,
     ):
         """
-        :param tk.Path audio_dir_path: path to raw audio files to be included
+        :param [tk.Path]|tk.Path audio_dir_path: list of paths or single path to raw audio files to be included
         :param str file_extension: file extension to look for in audio_dir_path
         :param float valid_percent: percentage of files to be in validation set
         :param int seed: random seed for splitting into train and valid set
         :param str|None path_must_contain: if set, path must contain this substring
             for a file to be included in the manifest
+        :param float upsampling_alpha: specifies how much to upsample low resource languages. Upsampling
+            calculation is done according to "Unsupervised Cross-Lingual Representation Learning for Speech
+            Recognition", see  arXiv:2006.13979v2
+        :param [str]|str|None manifest_audio_paths: explicitly specifies output paths in manifest for each
+            audio directory respectively. Allows to use different paths in the manifest than in the audio_dir_paths
         """
-        self.audio_dir_path = audio_dir_path
+        if isinstance(audio_dir_path, tk.Path):
+            self.audio_dir_paths = [audio_dir_path]
+        else:
+            assert isinstance(audio_dir_path, list)
+            self.audio_dir_paths = audio_dir_path
+        assert min([isinstance(path, tk.Path) for path in self.audio_dir_paths])
+
+        if len(self.audio_dir_paths) == 1:
+            assert (
+                upsampling_alpha is None
+            ), "Only one audio directory is given, upsampling not possible"
```

Comment:
I don't understand, why is upsampling not allowed if only one audio dir given?

Reply:
The upsampling is always done with respect to the amount of all languages. The less data given for a language the more it is upsampled compared to the other languages. If only one languages is given the upsampling factor computed is always one however for whatever alpha value is given (so no upsampling is done). I feel like it would be misleading to allow the user to specify an alpha value which does not influence the outcome of the algorithm at all. Also it would change the hash of the job without changing its output.

Do you think I should make this more clear in the assertion or in the doc-string?

Reply:
I would add to the doc string

Reply:
I added to the doc string